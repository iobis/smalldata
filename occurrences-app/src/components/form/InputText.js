import classNames from 'classnames'
import i18next from 'i18next'
import PropTypes from 'prop-types'
import React from 'react'
import { useTranslation } from 'react-i18next'

export default function InputText({ className, name, onChange, value }) {
  const { t } = useTranslation()
  const label = t(name + '.label')
  const placeholderKey = name + '.placeholder'
  const helpKey = name + '.help'

  return (
    <div className={classNames('column field', className)}>
      <label className="label">
        {label}
      </label>
      <input
        className="input"
        onChange={(e) => onChange(e.target.value)}
        placeholder={i18next.exists(placeholderKey) ? t(placeholderKey) : undefined}
        type="text"
        value={value}/>
      {i18next.exists(helpKey) ? <p className="help">{t(helpKey)}</p> : null}
    </div>
  )
}

InputText.propTypes = {
  className: PropTypes.string,
  name:      PropTypes.string.isRequired,
  onChange:  PropTypes.func.isRequired,
  value:     PropTypes.string
}
