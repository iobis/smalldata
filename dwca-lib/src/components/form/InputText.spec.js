import InputText from './InputText'
import React from 'react'
import { mount } from 'enzyme'

describe('InputText', () => {
  it('renders correctly', () => {
    expect(mount(createComponent())).toMatchSnapshot()
  })

  it('renders correctly if optional', () => {
    expect(mount(createComponent({ optional: true }))).toMatchSnapshot()
  })

  it('returns text value onChange', () => {
    const onChange = jest.fn()
    const wrapper = mount(createComponent({ onChange }))

    wrapper.find('input').simulate('change', { target: { value: 'new text value' } })

    expect(onChange).toBeCalledWith('new text value')
  })
})

function createComponent(props) {
  const defaultProps = {
    className: 'custom-class-name',
    name:      'field-name',
    onChange:  jest.fn(),
    value:     'value'
  }
  return <InputText {...defaultProps} {...props}/>
}
