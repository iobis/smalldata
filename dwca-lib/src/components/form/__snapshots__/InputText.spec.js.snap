// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText renders correctly 1`] = `
<InputText
  className="custom-class-name"
  name="field-name"
  onChange={[MockFunction]}
  value="value"
>
  <div
    className="column field custom-class-name"
  >
    <label
      className="label"
      htmlFor="field-name.label"
    >
      field-name.label
    </label>
    <input
      className="input"
      id="field-name.label"
      onChange={[Function]}
      type="text"
      value="value"
    />
  </div>
</InputText>
`;

exports[`InputText renders correctly if optional 1`] = `
<InputText
  className="custom-class-name"
  name="field-name"
  onChange={[MockFunction]}
  optional={true}
  value="value"
>
  <div
    className="column field custom-class-name"
  >
    <label
      className="label has-text-weight-normal"
      htmlFor="field-name.label"
    >
      field-name.label
    </label>
    <input
      className="input"
      id="field-name.label"
      onChange={[Function]}
      type="text"
      value="value"
    />
  </div>
</InputText>
`;
